vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -28,14 +48,14 @@ class DataManager:
     _cache = Cache(config={"CACHE_TYPE": "SimpleCache"})
 
     def __init__(self):
-        self.__lazy_data: Dict[DatasetName, pd_LazyDataFrame] = {}
+        self.__lazy_data: Dict[DatasetName, VizroDataSet] = {}
         self.__original_data: Dict[DatasetName, pd.DataFrame] = {}
         self.__component_to_original: Dict[ComponentID, DatasetName] = {}
         self._frozen_state = False
 
     # happens before dashboard build
     @_state_modifier
-    def __setitem__(self, dataset_name: DatasetName, data: Union[pd.DataFrame, pd_LazyDataFrame]):
+    def __setitem__(self, dataset_name: DatasetName, data: Union[pd.DataFrame, pd_LazyDataFrame, VizroDataSet]):
```

Comment:
```suggestion
    def __setitem__(self, dataset_name: DatasetName, data: Union[pd.DataFrame, pd_LazyDataFrame]):
```

Reply:
Since it'll be private, let's not expect the user to ever type `data_manager["iris"] = _Dataset` for now at least.